www/components/itinerum_tutorial/Tutorial.jsx

```diff
@@ -15,12 +15,21 @@ class TutorialComponent extends React.Component {
           lname: this.props.schema.lname
         }
       };
-      this.handleChange = this.handleChange.bind(this);
+      this.handlefnameChange = this.handlefnameChange.bind(this);
+      this.handlelnameChange = this.handlelnameChange.bind(this);
     }
-    handleChange(event) {
+    handlefnameChange(event) {
       this.setState({
         schema: {
           fname: event.target.value,
+          lname: this.state.schema.lname
+        }
+      });
```

Comment:
@kafitz that's super verbose! Can't you do something where you only set one field of the `schema` - eg. 

```
newSchema = schema.copy(); // not sure if you even need to copy or whether a reference is sufficient.
newSchema.fname = event.target.value; // note that in this, we only need to access fname in the handlefnameChange function
this.setState({schema: newSchema});
```

This doesn't remove the need for separate handlers, but as the number of fields in `schema` grows, this seems like it would be faster to code that copying 50 elements over.